src/azul/plugins/metadata/anvil/indexer/transform.py

```diff
@@ -418,7 +431,29 @@ def reconcile_inner_entities(cls,
                                  ) -> tuple[JSON, BundleFQID]:
         this_entity, this_bundle = this
         that_entity, that_bundle = that
-        return that if that_bundle.version > this_bundle.version else this
+        if this_entity.keys() == that_entity.keys():
+            return that if that_bundle.version > this_bundle.version else this
+        else:
+            assert entity_type == 'datasets', (entity_type, this, that)
+            expected_keys = cls.field_types()[entity_type].keys()
```

Comment:
I think this gets executed a lot (but correct me if I am wrong) and so we should cache the value on the RHS. I think a class attribute might do the trick.